lib/gems/pending/util/extensions/miq-benchmark.rb

```diff
@@ -22,15 +22,20 @@ def self.realtime_store(hash, key)
   # call completes.  If the hash already has a value for that key, the time is
   # accumulated.
   def self.realtime_block(key, &block)
-    outermost = !self.in_realtime_block?
+    outermost = !in_realtime_block?
     hash = current_realtime
-    self.current_realtime = hash if outermost
 
     begin
-      ret = realtime_store(hash, key, &block)
-      return ret, hash
+      self.current_realtime = hash if outermost
+      begin
+        ret = realtime_store(hash, key, &block)
+        return ret, hash
+      ensure
+        delete_current_realtime if outermost
+      end
+    # A second layer of protection in case TimeoutError struck before or after `begin..ensure`
     ensure
-      delete_current_realtime if outermost
+      delete_current_realtime if outermost && in_realtime_block?
```

Comment:
`outermost` is cached `!in_realtime_block?` at begining; `in_realtime_block?` is a call checking current state (original had `self.in_realtime_block?` which might have stressed it, drop because rubocop).
At this point it _should_ have original value restored, yes, so this && should never happen — unless Timeout::Error...

Rewritten as 2 branches — outermost or not — which IMO reads clearer? PTAL.